src/utils/templates/server-middleware.ts

```diff
@@ -0,0 +1,13 @@
+import type { Template } from '.'
+import { resolve } from 'pathe'
+
+const serverMiddleware: Template = ({ name, nuxtOptions }) => ({
+  path: nuxtOptions.future.compatibilityVersion === 3
+    ? resolve(nuxtOptions.srcDir, nuxtOptions.serverDir, 'middleware', `${name}.ts`)
+    : resolve(nuxtOptions.serverDir, 'middleware', `${name}.ts`),
```

Comment:
I think this is safe, as `serverDir` is fully resolved in compatibilityVersion: 4.

```suggestion
  path: resolve(nuxtOptions.srcDir, nuxtOptions.serverDir, 'middleware', `${name}.ts`)
```